lib/question_component/question.rb

```diff
@@ -7,9 +7,14 @@ class Question
     attribute :details, String
     attribute :tags, Array
     attribute :asked_time, Time
+    attribute :answers, Array, default: []
 
     def asked?
       !asked_time.nil?
     end
+
+    def answer(new_answer)
```

Comment:
Is this the best interface for the kind of behaviour

Reply:
I think you'd want each of the primitive attributes of a `Question::Answer` data structure to be arguments of this method. That way this class can encapsulate the constant look up of the subordinate "child" class.